pkg/cmd/roachtest/operations/backup_restore.go

```diff
@@ -69,64 +75,162 @@ outer:
+	done := ctx.Done()
+	conn, err := c.ConnE(ctx, o.L(), c.Node(1)[0])
 	if err != nil {
-		o.Fatal(err)
+		return err
 	}
-	restoredRow.Next()
-	if err := restoredRow.Scan(&restoreFingerprint); err != nil {
-		o.Fatal(err)
+	defer conn.Close()
+	for {
+		select {
+		case <-done:
+			return ctx.Err()
+		case <-succeededJobTick.C:
+			var status string
+			if err := conn.QueryRow(`SELECT status FROM [SHOW JOBS] WHERE job_type = 'RESTORE' ORDER BY created DESC LIMIT 1`).Scan(&status); err != nil {
+				return err
+			}
+			if status == string(jobs.StatusSucceeded) {
+				var externalBytes uint64
+				if err := conn.QueryRow(jobutils.GetExternalBytesForConnectedTenant).Scan(&externalBytes); err != nil {
+					return errors.Wrapf(err, "could not get external bytes")
+				}
+				if externalBytes != 0 {
+					return errors.Newf("not all data downloaded, %d external bytes still in cluster", externalBytes)
+				}
+				o.Status("download job completed")
+				return nil
+			} else if status == string(jobs.StatusRunning) {
+				o.Status("download job still running")
+			} else {
+				return errors.Newf("job unexpectedly found in %s state", status)
+			}
+		}
 	}
+}
 
-	if backupFingerprint != restoreFingerprint {
-		o.Fatalf("backup and restore fingerprints do not match: %d != %d", backupFingerprint, restoreFingerprint)
+func runBackupRestoreFn(
+	online bool, validate bool,
+) func(context.Context, operation.Operation, cluster.Cluster) registry.OperationCleanup {
+	return func(ctx context.Context, o operation.Operation, c cluster.Cluster) registry.OperationCleanup {
+		return runBackupRestore(ctx, o, c, online, validate)
 	}
-
-	return &backupRestoreCleanup{db: restoreDBName}
 }
 
 func registerBackupRestore(r registry.Registry) {
 	r.AddOperation(registry.OperationSpec{
-		Name:               "backup-restore/tpcc/district",
+		Name:               "backup-restore/tpcc/online=false/fingerprint=true",
+		Owner:              registry.OwnerDisasterRecovery,
+		Timeout:            96 * time.Hour,
+		CompatibleClouds:   registry.AllClouds,
+		CanRunConcurrently: registry.OperationCanRunConcurrently,
+		Dependencies:       []registry.OperationDependency{registry.OperationRequiresPopulatedDatabase},
+		Run:                runBackupRestoreFn(false, true),
+	})
+
+	r.AddOperation(registry.OperationSpec{
+		Name:               "backup-restore/tpcc/online=false/fingerprint=false",
+		Owner:              registry.OwnerDisasterRecovery,
+		Timeout:            24 * time.Hour,
+		CompatibleClouds:   registry.AllClouds,
+		CanRunConcurrently: registry.OperationCanRunConcurrently,
+		Dependencies:       []registry.OperationDependency{registry.OperationRequiresPopulatedDatabase},
+		Run:                runBackupRestoreFn(false, false),
+	})
+
+	r.AddOperation(registry.OperationSpec{
+		Name:               "backup-restore/tpcc/online=true/fingerprint=true",
+		Owner:              registry.OwnerDisasterRecovery,
+		Timeout:            96 * time.Hour,
```

Comment:
do we need to set these timeouts when we run the 100tb test? id like to see how long each operation takes. perhaps that's more of a core roachtest/operation change to make @itsbilal